packages/devextreme-react/src/core/use-option-scanning.ts

```diff
@@ -0,0 +1,159 @@
+import * as React from 'react';
+import {
+  Children,
+  ReactNode,
+  ReactElement,
+  useContext,
+  useRef,
+  useLayoutEffect,
+} from 'react';
+
+import { NestedOptionContext, NestedOptionContextContent, TemplateDiscoveryContext } from './helpers';
+import { ElementType, getElementType, IOptionElement } from './configuration/react/element';
+import { mergeNameParts } from './configuration/utils';
+import { separateProps } from './widget-config';
+import { IConfigNode, ITemplate } from './configuration/config-node';
+import { getAnonymousTemplate, getNamedTemplate } from './configuration/react/templates';
+import { ITemplateProps } from './template';
+
+function wrapTemplate(template: ITemplate): ITemplate {
+  return template.type === 'children' ? {
+    ...template,
+    content: React.createElement(
+      TemplateDiscoveryContext.Provider,
+      {
+        value: {
+          discoveryRendering: true,
+        },
+      },
+      template.content,
+    ),
+  } : template;
+}
+
+export function useOptionScanning(
+  optionElement: IOptionElement,
+  children: ReactNode,
+  templateContainer: HTMLDivElement,
+  parentUpdateToken: number,
+): [
+    IConfigNode,
+    NestedOptionContextContent,
+  ] {
+  const parentContext = useContext(NestedOptionContext);
+
+  const {
+    parentFullName,
+  } = parentContext;
+
+  const updateToken = Math.random();
+
+  const separatedValues = separateProps(
+    optionElement.props,
+    optionElement.descriptor.initialValuesProps,
+    optionElement.descriptor.templates,
+  );
+
+  const configCollections: Record<string, IConfigNode[]> = {};
+  const configCollectionMaps: Record<string, Record<string, number>> = {};
+  const configs: Record<string, IConfigNode> = {};
+  const templates: ITemplate[] = [];
+
+  Children.map(
```

Comment:
Return value is not used
```suggestion
  Children.forEach(
```